react/src/components/Table/Table.tsx

```diff
@@ -71,34 +72,103 @@ const resolveSex = (sexPhenotype: string) => {
-    return queryResult.flatMap(d => {
-        if (d.variant.callsets.length) {
-            //one row per individual per callset
-            return d.variant.callsets
-                .filter(cs => cs.individualId === d.individual.individualId)
-                .map(cs =>
-                    addAdditionalFieldsAndFormatNulls({
-                        ...cs.info,
-                        ...flattenBaseResults(d),
-                    })
-                );
+/* 1, Sort queryResult in ascending order according to variant's chromosome, ref, alt, start,end. 
+2, flatten data and compute values as needed 
+(note that column display formatting function should not alter values for ease of export) */
+const prepareData = (queryResult: VariantQueryDataResult[]): [ResultTableColumns[], number[]] => {
+    const sortedQueryResult = [...queryResult].sort(function (a, b) {
+        if (a.variant.referenceName < b.variant.referenceName) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref < b.variant.ref
+        ) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref === b.variant.ref &&
+            a.variant.alt < b.variant.alt
+        ) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref === b.variant.ref &&
+            a.variant.alt === b.variant.alt &&
+            a.variant.start < b.variant.start
+        ) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref === b.variant.ref &&
+            a.variant.alt === b.variant.alt &&
+            a.variant.start === b.variant.start &&
+            a.variant.end < b.variant.end
+        ) {
+            return -1;
         } else {
-            return addAdditionalFieldsAndFormatNulls(flattenBaseResults(d));
+            return 1;
         }
     });
+
+    var result: Array<ResultTableColumns> = [];
+    var uniqueVariantIndices: Array<number> = []; // contains indices of first encountered rows that represent unique variants.
+    const n = sortedQueryResult.length;
+    var currVariant = { chromosome: 'temp', ref: 'temp', alt: 'temp', start: 0, end: 0 };
+    var currUniqueId = 0;
+    var currRowId = 0;
+
+    for (let i = 0; i < n; i++) {
+        const d = sortedQueryResult[i];
+        if (
+            d.variant.referenceName !== currVariant['chromosome'] ||
+            d.variant.ref !== currVariant['ref'] ||
+            d.variant.alt !== currVariant['alt'] ||
+            d.variant.start !== currVariant['start'] ||
+            d.variant.end !== currVariant['end']
+        ) {
+            currUniqueId += 1;
+            uniqueVariantIndices.push(currRowId);
+            currVariant['chromosome'] = d.variant.referenceName;
```

Comment:
Can you explain why we would need to assign values to `currVariant` here, since it's not used anywhere else after this code block?

Reply:
Actually, I got it. You are trying to reuse the value for comparison in the `if` condition above to determine if a variant is unique.

Reply:
yes, that's right.